tests/ui/manual_main_separator_str.fixed

```diff
@@ -0,0 +1,14 @@
+// run-rustfix
+
+#![allow(unused)]
+#![warn(clippy::manual_main_separator_str)]
+
+use std::path::MAIN_SEPARATOR;
+
+fn main() {
```

Comment:
What about `let`s without a given type? Or other expressions that are not direct local bindings?

Reply:
I have added new tests, such as initializing a `struct` field with this expression or passing it as a function argument.

I've also made it work when it is a method receiver (see the test with `encode_utf16()`), without an explicit `&`.